### PR TITLE
feat: bns support and registration for connect

### LINKED ISSUE #736: [Blockstack Connect] Update default subdomain registrar to non-PBC host and restore username paths
The URL value for the default registrar used for username registration by Blockstack Connect (`app.blockstack.org`) should be updated once the registrar is hosted by a non-PBC entity such as the Foundation.

Then all existing username flows should be restored having been disabled for Stacks 2.0 launch.

### LINKED ISSUE #946: Restore username registration functionality for production apps
We had to disable username registration functionality for production apps upon the release of Stacks 2.0 since the underlying subdomain registrar powering both `app.blockstack.org` and `browser.blockstack.org` was not updated in time.

Several app developers have disabled their onboarding flows entirely while they wait for us to restore this functionality. 

This issue is meant to track the entirety of this work in one place, linking to other issues and PRs as relevant.

# Steps towards restoration

- [x] [Update subdomain registrar to Stacks 2.0](https://github.com/blockstack/subdomain-registrar/pull/42) (Tintash)
- [x] [Update ENV variable handling in support of hosting needs](https://github.com/blockstack/subdomain-registrar/issues/44) (UserX @ Hiro PBC)
- [x] [Resolve issues with Atlas for testnet and mainnet](https://github.com/blockstack/ux/issues/946#issuecomment-778112118) (Blockchain @ Hiro PBC)
- [x] [Host registrar against testnet](https://github.com/blockstackpbc/devops/issues/593) (Devops @ Hiro PBC)
- [x] Remove core.blockstack.org dependency in favor of Stacks Blockchain API for [Stacks Wallet, Blockstack Connect](https://github.com/blockstack/ux/issues/956) and [Stacks.js](https://github.com/blockstack/stacks.js/issues/945) (UserX @ Hiro PBC)
- [x] [Deploy Atlas fixes to mainnet](https://github.com/blockstack/ux/issues/946#issuecomment-778112118) (Blockchain @ Hiro PBC)
- [x] [Host registrar against mainnet](https://github.com/blockstack/subdomain-registrar/issues/43) (Devops @ Hiro PBC for Stacks Foundation)
- [x] Deploy new version of [Blockstack Connect (`app.blockstack.org`)](https://github.com/blockstack/ux/issues/736) with username registration restored using registrar newly hosted by Stacks Foundation (UserX @ Hiro PBC)
- [ ] Reenable onboarding flows for apps (App developers)

## Contributing parties

- Tintash team: @asimm241
- Stacks Foundation: @blocks8 @jcnelson
- Hiro PBC: @andresgalante @diwakergupta  @gina @hstove @wileyj  @markmhx 
- App developers: @pradel @raj015 @sdsantos @wilsonbright

### LINKED ISSUE #956: Integrate Stacks Blockchain API for username-related functionality
For both products:

- [ ] Stacks Wallet
- [ ] Blockstack Connect (app.blockstack.org)

This is presumably possible now that https://github.com/blockstack/stacks-blockchain-api is merged into the API.

### PR DESCRIPTION
**Pull request by hstove**
> Try out this version of the Stacks Wallet - try the [hosted version](https://pr-1103.app.stacks.engineering), or the [Stacks testnet demo app](https://pr-1103.testnet-demo.stacks.engineering)

Fixes #736, fixes #956, fixes #946

This PR does the following:
- Adds a new subdomain registrar to our configuration
- Sets this new registrar for app.blockstack.org registrations
- re-enables subdomain registrations on app.blockstack.org

Currently only a testnet subdomain registrar is configured. This upgrade must go with https://github.com/blockstack/stacks.js/pull/984 in order for apps to properly handle the upgrade.

- [ ] Add production registrar (once available)

### CODE REVIEW

**Comment by markmhendrickson** [comment]
@hstove Should we take this out of draft mode and move into QA?

And do we want to modify the testnet demo app to provide two auth options / buttons: one uses the `registerSubdomain` flag and one that doesn't, for testing both cases?

**Comment by markmhendrickson** [comment]
I was able to register a username with the Testnet Demo app here: 

https://stacks-node-api.testnet.stacks.co/v1/names/awegawegawegawegawe.test-registrar.id

**Comment by markmhendrickson** [comment]
@hstove Would it be easy enough to sneak in a "Skip" option on the username screen here?

@wilsonbright just requested it given their recent experience with improved conversion without required username registration.

If not, we can break it out into its own issue and track there.

**Review by markmhendrickson — APPROVED** [review]

File: packages/keychain/src/profiles.ts [hunk 1]
```diff
@@ -102,22 +120,25 @@ const sendUsernameToRegistrar = async ({
...
   });
 
-  const requestHeaders = {
+  const headers = new Headers({
     Accept: 'application/json',
     'Content-Type': 'application/json',
-  };
+  });
+  if (apiKey) {
+    headers.set('Authorization', `bearer ${apiKey}`);
```
Comment (kyranjamie) [inline packages/keychain/src/profiles.ts:141]:
```suggestion
    headers.set('Authorization', `Bearer ${apiKey}`);
```
Isn't this typically capitalised?

**Comment by markmhendrickson** [comment]
Registration worked for me:

image

**Comment by markmhendrickson** [comment]
@kyranjamie and/or @aulneau can you code review this soon? We should release as soon as possible.

@hstove is there any other change necessary, or will we be good to deploy once code review is complete?

File: packages/app/package.json [hunk 2]
```diff
@@ -100,11 +100,9 @@
...
     "jest-circus": "^25.1.0",
     "jest-junit": "^12.0.0",
-    "playwright": "^1.6.2",
-    "playwright-chromium": "^1.6.2",
-    "playwright-core": "^1.6.2",
-    "playwright-firefox": "^1.6.2",
-    "playwright-webkit": "^1.6.2",
+    "playwright": "^1.9.2",
+    "playwright-chromium": "^1.9.2",
+    "playwright-core": "^1.9.2",
```
Comment (kyranjamie) [inline packages/app/package.json:105]: There's a audit job right?

File: packages/app/src/common/constants.ts [hunk 3]
```diff
@@ -2,12 +2,13 @@ import { Subdomains } from '@stacks/keychain';
...
 
 if (document?.location.origin.includes('localhost')) {
-  Subdomain = Subdomains.TEST;
+  Subdomain = Subdomains.TEST_V2;
 }
 
 export const transition = 'all .2s cubic-bezier(.215,.61,.355,1)';
 
-export const USERNAMES_ENABLED = process.env.USERNAMES_ENABLED === 'true';
+// export const USERNAMES_ENABLED = process.env.USERNAMES_ENABLED === 'true';
```
Comment (kyranjamie) [inline packages/app/src/common/constants.ts]:
```suggestion
```

File: packages/keychain/src/profiles.ts [hunk 4]
```diff
@@ -46,16 +47,39 @@ const DEFAULT_PROFILE_FILE_NAME = 'profile.json';
...
 
-export const registrars = {
+interface Registrars {
+  [key: string]: {
+    registerUrl: string;
+    apiUrl: string;
+    apiKey?: string;
+    addressVersion: number;
+  };
+}
```
Comment (kyranjamie) [inline packages/keychain/src/profiles.ts]:
```suggestion
interface Registrar {
  registerUrl: string;
  apiUrl: string;
  apiKey?: string;
  addressVersion: number;
};

type Registrars = Record<string, Registrar>;
``` 

Pref type structure

**Review by kyranjamie — APPROVED** [review]

File: packages/keychain/src/profiles.ts (hunk 1 above)
Comment (hstove) [reply packages/keychain/src/profiles.ts:141]: Typically, yeah. But the registrar expects lowercase 🤷🏼 .

File: packages/app/package.json (hunk 2 above)
Comment (hstove) [reply packages/app/package.json:105]: Not on this branch

**Pull request merged**